state/status.go

```diff
@@ -191,6 +191,17 @@ func probablyUpdateStatusHistory(st *State, globalKey string, doc statusDoc) {
 	}
 }
 
+func eraseStatusHistory(st *State, globalKey string) error {
+	history, closer := st.db().GetCollection(statusesHistoryC)
+	defer closer()
+	historyW := history.Writeable()
+
+	if _, err := historyW.RemoveAll(bson.D{{"globalkey", globalKey}}); err != nil {
```

Comment:
Sometimes I really dislike the "inject model-uuid" abstraction we put in place. Certainly is hard when you're jumping between Juju code and trying to run it live.
Having a "globalKey" string constant here feels easy to get wrong.
Maybe if we put the string next to the struct definition? It would be great if we could do something like:
type historyDoc struct {
  globalKey `json:$THEKEY`
}
...
 history.RemoveAll($THEKEY)

I believe you can do
reflect.Type(historyDoc).FieldByName("globalKey").Tag().Get("bson")
but that certainly doesn't fit the "make it easy to keep them in sync" and you still have that "globalKey" constant (though its naming the attribute rather than a database field).

Anyway, this seems ok, maybe slightly better with a string const next to the struct definition.